packages/client/src/components/interface/DownloadButton.tsx

```diff
@@ -90,37 +273,67 @@ function DownloadButtonComponent(props: DownloadButtonProps & HOCProps) {
     )
   }
 
-  if (
-    status === DOWNLOAD_STATUS.FAILED ||
-    status === DOWNLOAD_STATUS.FAILED_NETWORK
-  ) {
-    return (
-      <StatusIndicator className={className} id={`${id}-download-failed`}>
-        <DownloadAction
-          id={`${id}-icon`}
-          onClick={(e) => {
-            initiateDownload()
-            e.stopPropagation()
-          }}
-          icon={() => <Download isFailed={true} />}
-        />
-      </StatusIndicator>
-    )
-  }
-
   return (
-    <DownloadAction
-      id={`${id}-icon`}
-      onClick={(e) => {
-        initiateDownload()
-        e.stopPropagation()
-      }}
-      icon={() => <Download />}
-      className={className}
-    />
+    <>
+      <DownloadAction
+        id={`${id}-icon`}
+        onClick={onClickDownload}
+        className={className}
+      >
+        {status === DOWNLOAD_STATUS.DOWNLOADED ? (
+          <Downloaded />
+        ) : assignment && assignment.userId != userId ? (
+          <AvatarVerySmall
+            avatar={{
+              data: `${window.config.API_GATEWAY_URL}files/avatar/${assignment.userId}.jpg`,
+              type: 'image/jpeg'
+            }}
+          />
+        ) : (
+          <Download
+            isFailed={
+              status === DOWNLOAD_STATUS.FAILED ||
+              status === DOWNLOAD_STATUS.FAILED_NETWORK
+            }
+          />
+        )}
+      </DownloadAction>
+      {assignModal !== null && (
+        <ResponsiveModal
+          show
+          title={intl.formatMessage(assignModal.title)}
+          actions={assignModal.actions.map((action) =>
+            renderModalAction(action, intl)
+          )}
+          autoHeight
+          responsive={false}
+          preventClickOnParent
```

Comment:
Had to introduce this `preventClickOnParent` prop to prevent click on row item, as the modal is rendered as a child of row, row click event handler is triggered when clicked anywhere the modal